platform/android/detect.py

```diff
@@ -41,17 +41,17 @@ def get_doc_path():
     return "doc_classes"
 
 
-# Return the ANDROID_SDK_ROOT environment variable.
+# Return the ANDROID_HOME environment variable.
 def get_env_android_sdk_root():
-    return os.environ.get("ANDROID_SDK_ROOT", -1)
+    return os.environ.get("ANDROID_HOME", os.environ.get("ANDROID_SDK_ROOT"))
```

Comment:
Should the fallback also have a fallback? Not sure what `-1` achieved, should maybe be an empty string? Otherwise it may be None and lead to errors.

Reply:
I thought `None` was preferred, so we can do things like `if get_env_android_sdk_root():`.

I'm no python expert so happy to defer to the recommended practice.

Reply:
Well currently the only use of `get_env_android_sdk_root()` is here:
```
("ANDROID_HOME", "Path to the Android SDK", get_env_android_sdk_root()),
```

Which means `env["ANDROID_HOME"]` would be `None`, and thus code like `env["ANDROID_HOME"] + "/ndk/" + get_ndk_version()` will fail.

Reply:
Added the empty string as a fallback. We have various `os.path.exists(...)` checks so they should quickly detect if `env["ANDROID_HOME"]` has an invalid value / path.